index.js

```diff
@@ -85,14 +85,36 @@ function checkMessagePathQueryParameter(req, res, next) {
 }
 
 function checkTopicQueryParameter(req, res, next) {
-
     if (req.query.topic) {
         req.body.topic = req.query.topic;
     }
+    next();
+}
 
+function checkQoSParameter(req,res,next){
+  if (req.body.qos){
+    req.body.qos = 0;
+    next();
+  }
+  else {
+    let qos = parseInt(req.body.qos, 10);
+    if (qos < 0 || qos > 2){
+      res.status(400).send("QoS range is not in range 0-2");
+  }
```

Comment:
Please try to run this thing locally first. I think you missed the } here.

Reply:
Did it, didnt throw any errors on my pc